core/src/main/scala/org/elasticmq/metrics/MBeanMetrics.scala

```diff
@@ -0,0 +1,90 @@
+package org.elasticmq.metrics
+
+import java.util.concurrent.{Executors, TimeUnit}
+
+import akka.actor.ActorRef
+import akka.util.Timeout
+import javax.management.openmbean._
+import org.elasticmq.QueueStatistics
+import org.elasticmq.metrics.QueuesMetrics.{queueDataNames, queueStatisticsCompositeType, queueStatisticsTabularType}
+import org.elasticmq.util.NowProvider
+
+import scala.concurrent.{Await, ExecutionContext}
+
+trait QueuesMetricsMBean {
+  def getQueueNames: Array[String]
+
+  def getNumberOfMessagesInQueue(queueName: String): CompositeData
+
+  def getNumberOfMessagesForAllQueues: TabularData
+}
+
+class QueuesMetrics(queueManagerActor: ActorRef) extends QueuesMetricsMBean {
+
+  implicit val timeout: Timeout = Timeout(21, TimeUnit.SECONDS)
+  implicit val ec: ExecutionContext = ExecutionContext.fromExecutorService(Executors.newFixedThreadPool(2))
```

Comment:
maybe a cached thread pool would be sufficient, to only start these threads when there's demand?

Reply:
Yup, it should be better here, I have changed it to cached one